include/pybind11/detail/class.h

```diff
@@ -531,6 +531,9 @@ extern "C" inline int pybind11_set_dict(PyObject *self, PyObject *new_dict, void
 extern "C" inline int pybind11_traverse(PyObject *self, visitproc visit, void *arg) {
     PyObject *&dict = *_PyObject_GetDictPtr(self);
     Py_VISIT(dict);
+#if PY_VERSION_HEX >= 0x03090000
+    Py_VISIT(Py_TYPE(self));
```

Comment:
Optional: I'd add this comment (your link from the PR description) so the next person to look here has an easier time understanding this code:
```
// https://docs.python.org/3/c-api/typeobj.html#c.PyTypeObject.tp_traverse
```